ThriftHandler/MapDHandler.cpp

```diff
@@ -720,35 +720,79 @@ void MapDHandler::get_completion_hints(std::vector<TCompletionHint>& hints,
                                        const TSessionId& session,
                                        const std::string& sql,
                                        const int cursor) {
-  const auto last_word = find_last_word_from_cursor(sql, cursor < 0 ? sql.size() : cursor);
+  get_completion_hints_unsorted(hints, session, sql, cursor);
+  // Sort the hints by category, from COLUMN (most specific) to KEYWORD.
+  std::sort(hints.begin(), hints.end(), [](const TCompletionHint& lhs, const TCompletionHint& rhs) {
+    return lhs.type < rhs.type;
+  });
+}
```

Comment:
oh, I like this hint sort order.  this is something that @tmostak and I had also talked about 👌